src/net/client/mod.rs

```diff
@@ -67,27 +65,31 @@ impl MpcHelperClient {
         Ok(Self::new(addr.parse()?))
     }
 
-    fn build_uri<T>(&self, p_and_q: T) -> Result<Uri, Error>
-    where
-        PathAndQuery: TryFrom<T>,
-        <PathAndQuery as TryFrom<T>>::Error: Into<axum::http::Error>,
-    {
-        Ok(uri::Builder::new()
-            .scheme(self.scheme.clone())
-            .authority(self.authority.clone())
-            .path_and_query(p_and_q)
-            .build()?)
-    }
-
     /// Responds with whatever input is passed to it
     /// # Errors
     /// If the request has illegal arguments, or fails to deliver to helper
-    pub async fn echo(&self, s: &str) -> Result<Vec<u8>, Error> {
-        let uri = self.build_uri(http_serde::echo::uri(s))?;
+    pub async fn echo(&self, s: &str) -> Result<String, Error> {
+        const FOO: &str = "foo";
```

Comment:
It seems that it's been reviewed already. Is it the case? can you maybe rebase it on top of e2e branch?

Reply:
it's rebased now